ffs/minerselector/reptop/reptop.go

```diff
@@ -29,24 +29,36 @@ func New(rm *reputation.Module, ai *askRunner.Runner) *RepTop {
 // GetMiners returns n miners using the configured Reputation Module and
 // Ask Index.
 func (rt *RepTop) GetMiners(n int, f ffs.MinerSelectorFilter) ([]ffs.MinerProposal, error) {
-	ms, err := rt.rm.QueryMiners(n, f.ExcludedMiners, f.CountryCodes, f.TrustedMiners)
+	if n < 1 {
+		return nil, fmt.Errorf("the number of miners should be greater than zero")
+	}
+	ms, err := rt.rm.QueryMiners(f.ExcludedMiners, f.CountryCodes, f.TrustedMiners)
```

Comment:
`rt.rm.QueryMiners` now returns a list of _all_ miners that match excluded, countries, and trusted. We don't want to limit the length of that list because we can't know if we'll use any miner in the list until we inspect their asks and compare to max price.

Reply:
I see. I think eventually we should give more thought about miner-selectors... since returning the complete list sounds like might give some problems if there're a lot of miners. Definitely not something with a good return on time investment now, but to food for thought.

Reply:
yea, agreed.